### PR TITLE
Updated eslint and added react plugin (fixes #26)

### LINKED ISSUE #26: Upgrade to eslint 0.20.0
I wouldn't expect any major issues.

### PR DESCRIPTION
**Pull request by job13er**
This change updates the `peerDependencies` to `eslint@0.20.0` and introduces
the `eslint-plugin-react` for better React.js support.

### CODE REVIEW

**Comment by job13er** [comment]
I did verify that using this version of `beaker` with no `.eslintrc` changes still passed `make lint` on `cy-flex-grid` so I think it's safe to call it a minor change.

File: .eslintrc [hunk 1]
```diff
@@ -71,6 +74,22 @@
...
+        "react/jsx-uses-react": 1,
+        "react/jsx-uses-vars": 1,
+        "react/no-did-mount-set-state": 1,
+        "react/no-did-update-set-state": 1,
+        "react/no-multi-comp": 1,
+        "react/no-unknown-property": 1,
+        "react/prop-types": 1,
+        "react/react-in-jsx-scope": 1,
+        "react/self-closing-comp": 1,
+        "react/wrap-multilines": 1,
```
Comment (job13er) [inline .eslintrc]: Defaulting these to warnings to avoid breaking changes. Next major bump may make them errors.

**Comment by job13er** [comment]
@psbanka Also confirmed we can remove those darn `/* eslint-disable no-unused-vars */` lines with this updated `beaker` and `.eslintrc`

**Comment by psbanka** [comment]
Looks great! :+1:

**Pull request merged**